content/articles/how-to-build-discord-bot-with-javascript/index.md

```diff
@@ -0,0 +1,163 @@
+### introduction
+Discord is a robust communication app used by a large community of gamers, developers, and cryptocurrency enthusiasts. Admins can add a plethora of unique Discord bots to their servers. There are also powerful APIs for creating Discord bots for those who prefer to take matters into their own hands.
+With Discord.js, it is relatively simple to create a Discord bot using Javascript. In this Discord bot tutorial, we'll go over everything you need to know to get a bot up and running on the Discord server.
+### Table of content
+- [introduction](#introduction)
+- [Table of content](#table-of-content)
+- [Requirements](#requirements)
+- [Add New Discord Application](#add-new-discord-application)
+- [Create New Node.js Project](#create-new-nodejs-project)
+- [Install dependencies.](#install-dependencies)
+- [Building our bot](#building-our-bot)
+- [Create the bot file](#create-the-bot-file)
+- [Have bot reply to messages](#have-bot-reply-to-messages)
+- [Summary](#summary)
+- [Conclusion](#conclusion)
+
+### Requirements
+1. wifi connection
+2. discord account
+3. text editor
+4. Node v17.1.0 (latest version)
+### Add New Discord Application
+visit https://discord.com/developers/applications. 
+1. Click on the New Application button.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/new.png)
+   After clicking the button a dialog box will pop up which will ask you to name your bot, In my case, I will name mine 'mybot'.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/dialog.png)
+2. Click the bot option, then Add bot button.
+This dialog will pop up.
+![](/education-engineering/how-to-build-discord-bot-with-javascript/yes.png)
+Confirm the creation of the new application and let's proceed.
+3. Here, we will give our bot extra permission to improve its interactivity with users. Click the OAuth2 menu and mark as shown below. Copy the link that appears in the text box.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/bot.png)
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/permission.png)
+4. Open a new tab and paste your link(copied in step 3).
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/link.png)
+5. select your server, hit authorize.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/authorize.png)
+in case you don't have a server. Login to your discord account and create one. 
+### Create New Node.js Project
+First, we need to set up our project, do the following:
+1. Create a folder where you want to store your files.
+2. Initialize a new node.js project, by opening the folder in your text editor.
+3. Install node.js/update if necessary.
+ ### Install dependencies.
+Dependencies are packages that are required by other packages to work. Here we will be using ``discord.js``, ``dotenv`` and ``nodemon``.

```

Comment:
Add a space between one heading and the next. This should be done across the article to make it readable.